### ISSUE TITLE
Poor performance in  datasets（5way1shot)

### ISSUE THREAD

**Issue by linjieb** [zhangdoudou123/SemFew#21]
1.resnet:
CIFAR-FS:79.61+0.80%
FC100:45.86+0.75%
MiniImageNet:72.70+0.67%
2.swin:
CIFAR-FS:81.41+0.76%
FC100:46.66+0.69%
MiniImageNet:73.04+0.75%
The above is my reproduction of the results of the three datasets, the effect is very poor, can you reproduce the same results as the original text, if it is convenient, discuss it with me.

**Comment by LiuMeilin7195** [comment]
The "mix acc" in test.txt is the final accuracy, which reflects the reproducibility of the results, not the "gen acc". I checked the test.txt you provided in another issue.

**Comment by linjieb** [comment]
> The "mix acc" in test.txt is the final accuracy, which reflects the reproducibility of the results, not the "gen acc". I checked the test.txt you provided in another issue.

Oh,yes!May be you are right,it is my results,but the fc100 in resnet is not good and I can't finish the TieredImageNet becuase of lacking of the file "find_labels.csv",and you?

![Image](https://github.com/user-attachments/assets/8b835a1b-3243-495a-9740-9ec5a6ee47ed)

**Comment by LiuMeilin7195** [comment]
I also could not find the file "find_labels.csv." My solution was to rewrite  dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.

**Comment by 777aaa** [comment]
> I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.

Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results  in MiniImageNet also not good!

**Comment by linjieb** [comment]
> I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.

Thanks!3 epochs taking 5 hours may be innormal I think.Then,would you like to continue to make it as your baseline?If so,we can keep in contact!

**Comment by LiuMeilin7195** [comment]
> > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> 
> Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results in MiniImageNet also not good!

Rewrite dataloader "tiered_imagenet.py" according to the dataset format, and "tiered_imagenet.py" is located in the directory: SemFew/data/

**Comment by LiuMeilin7195** [comment]
> > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> 
> Thanks!3 epochs taking 5 hours may be innormal I think.Then,would you like to continue to make it as your baseline?If so,we can keep in contact!

I will refer to the reproduction results of TieredImageNet to decide whether to use it as baseline. Currently the reproduction results on the Swin Transformer backbone for the three datasets differ from the paper by less than 1%. Let's stay in touch!

**Comment by 777aaa** [comment]
> > > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> > 
> > 
> > Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results in MiniImageNet also not good!
> 
> Rewrite dataloader "tiered_imagenet.py" according to the dataset format, and "tiered_imagenet.py" is located in the directory: SemFew/data/

Thanks for reply!I do not know how to rewrite the dataloader "tiered_imagenet.py" according to the dataset format.Could your share the code ? Thanks a lot!

**Comment by LiuMeilin7195** [comment]
> > > > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> > > 
> > > 
> > > Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results in MiniImageNet also not good!
> > 
> > 
> > Rewrite dataloader "tiered_imagenet.py" according to the dataset format, and "tiered_imagenet.py" is located in the directory: SemFew/data/
> 
> Thanks for reply!I do not know how to rewrite the dataloader "tiered_imagenet.py" according to the dataset format.Could your share the code ? Thanks a lot!

The directory format of the TieredImageNet dataset I downloaded is as follows:

**Comment by LiuMeilin7195** [comment]
> > > > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> > > 
> > > 
> > > Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results in MiniImageNet also not good!
> > 
> > 
> > Rewrite dataloader "tiered_imagenet.py" according to the dataset format, and "tiered_imagenet.py" is located in the directory: SemFew/data/
> 
> Thanks for reply!I do not know how to rewrite the dataloader "tiered_imagenet.py" according to the dataset format.Could your share the code ? Thanks a lot!

The directory format of the TieredImageNet dataset I downloaded is as follows:
TieredImageNet:
----train
--------n06785654
-------------n0678565400000001.jpg
-------------n0678565400000002.jpg
-------------...
----val(Same as the folder train)
----test(Same as the folder train)

My code:
[tiered_imagenet.txt](https://github.com/user-attachments/files/19157466/tiered_imagenet.txt)

If you obtaine reproduction results on the TieredImageNet dataset, please share them with us. Thank you!

**Comment by 777aaa** [comment]
> > > > > I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.
> > > > 
> > > > 
> > > > Hey,I also met the trouble,Could your share the solution to tackle the lack of "find_labels.csv"? And my results in MiniImageNet also not good!
> > > 
> > > 
> > > Rewrite dataloader "tiered_imagenet.py" according to the dataset format, and "tiered_imagenet.py" is located in the directory: SemFew/data/
> > 
> > 
> > Thanks for reply!I do not know how to rewrite the dataloader "tiered_imagenet.py" according to the dataset format.Could your share the code ? Thanks a lot!
> 
> The directory format of the TieredImageNet dataset I downloaded is as follows: TieredImageNet: ----train --------n06785654 -------------n0678565400000001.jpg -------------n0678565400000002.jpg -------------... ----val(Same as the folder train) ----test(Same as the folder train)
> 
> My code: [tiered_imagenet.txt](https://github.com/user-attachments/files/19157466/tiered_imagenet.txt)
> 
> If you obtaine reproduction results on the TieredImageNet dataset, please share them with us. Thank you!

Thanks! I will share the result when i finsh the reproduction.

**Comment by MountainChenCad** [comment]
> > The "mix acc" in test.txt is the final accuracy, which reflects the reproducibility of the results, not the "gen acc". I checked the test.txt you provided in another issue.
> 
> Oh,yes!May be you are right,it is my results,but the fc100 in resnet is not good and I can't finish the TieredImageNet becuase of lacking of the file "find_labels.csv",and you?
> 
> ![Image](https://github.com/user-attachments/assets/8b835a1b-3243-495a-9740-9ec5a6ee47ed)

I also encounter these issues! Have you figured out why?

Here are my test.txt.
CIFAR-FS
```
2025-03-14 20:22:19,255 - test.py[line:51] - INFO: {'test_batch': 600, 'shot': 1, 'query': 15, 'test_way': 5, 'center': 'mean', 'seed': 13, 'feat_size': 640, 'semantic_size': 512, 'num_workers': 8, 'mode': 'clip', 'text_type': 'gpt', 'dataset': 'CIFAR-FS', 'backbone': 'resnet', 'model_path': './checkpoints/ResNet-CIFAR-FS.pth', 'work_dir': 'resnet_CIFAR-FS_clip_gpt_mean_1'}
2025-03-14 20:22:23,556 - test.py[line:107] - INFO: best epoch: 28 77.975557
2025-03-14 20:22:23,556 - test.py[line:108] - INFO: best k: 0.180000
2025-03-14 20:22:38,403 - test.py[line:144] - INFO: max |k:             0.18 |mix acc: 82.78+0.71% |gap: 11.19
2025-03-14 20:22:38,404 - test.py[line:146] - INFO: ACC:|proto acc: 71.59+0.88% |gen acc: 79.74+0.80%
```
FC100
```
2025-03-15 07:59:31,195 - test.py[line:51] - INFO: {'test_batch': 600, 'shot': 1, 'query': 15, 'test_way': 5, 'center': 'mean', 'seed': 13, 'feat_size': 640, 'semantic_size': 512, 'num_workers': 8, 'mode': 'clip', 'text_type': 'gpt', 'dataset': 'FC100', 'backbone': 'resnet', 'model_path': './checkpoints/ResNet-FC100.pth', 'work_dir': 'resnet_FC100_clip_gpt_mean_1'}
2025-03-15 07:59:33,727 - test.py[line:107] - INFO: best epoch: 17 45.348890
2025-03-15 07:59:33,727 - test.py[line:108] - INFO: best k: 0.240000
2025-03-15 07:59:45,975 - test.py[line:144] - INFO: max |k:             0.24 |mix acc: 51.16+0.74% |gap: 8.19
2025-03-15 07:59:45,975 - test.py[line:146] - INFO: ACC:|proto acc: 42.97+0.74% |gen acc: 46.63+0.75%
```
MiniImageNet
```
2025-03-14 16:54:59,165 - test.py[line:51] - INFO: {'test_batch': 600, 'shot': 1, 'query': 15, 'test_way': 5, 'center': 'mean', 'seed': 13, 'feat_size': 640, 'semantic_size': 512, 'num_workers': 8, 'mode': 'clip', 'text_type': 'gpt', 'dataset': 'MiniImageNet', 'backbone': 'resnet', 'model_path': './checkpoints/ResNet-MiniImageNet.pth', 'work_dir': 'resnet_MiniImageNet_clip_gpt_mean_1'}
2025-03-14 16:55:39,826 - test.py[line:51] - INFO: {'test_batch': 600, 'shot': 1, 'query': 15, 'test_way': 5, 'center': 'mean', 'seed': 13, 'feat_size': 640, 'semantic_size': 512, 'num_workers': 8, 'mode': 'clip', 'text_type': 'gpt', 'dataset': 'MiniImageNet', 'backbone': 'resnet', 'model_path': './checkpoints/ResNet-MiniImageNet.pth', 'work_dir': 'resnet_MiniImageNet_clip_gpt_mean_1'}
2025-03-14 16:55:41,644 - test.py[line:107] - INFO: best epoch: 42 93.695558
2025-03-14 16:55:41,644 - test.py[line:108] - INFO: best k: 0.000000
2025-03-14 16:56:47,607 - test.py[line:144] - INFO: max |k:              0.0 |mix acc: 93.18+0.36% |gap: 16.45
2025-03-14 16:56:47,607 - test.py[line:146] - INFO: ACC:|proto acc: 76.73+0.88% |gen acc: 93.18+0.36%
2025-03-15 07:13:37,298 - test.py[line:51] - INFO: {'test_batch': 600, 'shot': 1, 'query': 15, 'test_way': 5, 'center': 'mean', 'seed': 13, 'feat_size': 640, 'semantic_size': 512, 'num_workers': 8, 'mode': 'clip', 'text_type': 'gpt', 'dataset': 'MiniImageNet', 'backbone': 'resnet', 'model_path': './checkpoints/ResNet-MiniImageNet.pth', 'work_dir': 'resnet_MiniImageNet_clip_gpt_mean_1'}
2025-03-15 07:13:39,857 - test.py[line:107] - INFO: best epoch: 49 83.386669
2025-03-15 07:13:39,858 - test.py[line:108] - INFO: best k: 0.240000
2025-03-15 07:14:46,104 - test.py[line:144] - INFO: max |k:             0.24 |mix acc: 76.03+0.65% |gap: 11.40
2025-03-15 07:14:46,104 - test.py[line:146] - INFO: ACC:|proto acc: 64.63+0.81% |gen acc: 72.50+0.67%
```

**Comment by wyb1018666** [comment]
> I also could not find the file "find_labels.csv." My solution was to rewrite dataloader "tiered_imagenet.py" according to the dataset format, which eliminates the need for the "find_labels.csv" file. As for the TieredImageNet dataset, I found a link to download it myself. The final reproduction results for TieredImageNet are not yet available, as training for 3 epochs took 5 hours, so I cannot guarantee that the solution I mentioned above will be effective. My results in FC100 with ResNet12 backbone also not good.

Hello, I'm currently planning to use this paper as my benchmark, and I just started reproducing the code for this experiment today. However, there's no link to the tieredImageNet dataset. Could you please tell me how to handle this? If you could provide a download link, I will reproduce it according to your method of rewriting tiered_imagenet.py. Thank you!
